crates/rpc/rpc/src/eth/api/transactions.rs

```diff
@@ -296,6 +297,48 @@ where
             status_code: if receipt.success { Some(U64::from(1)) } else { Some(U64::from(0)) },
         };
 
+        // get all receipts for the block
+        let all_receipts = match self.client().receipts_by_block((meta.block_number).into())? {
+            Some(recpts) => recpts,
+            None => return Err(EthApiError::UnknownBlockNumber),
+        };
+
+        // get the previous transaction cumulative gas used
+        let prev_tx_cumulative_gas_used = match all_receipts.get((meta.index - 1) as usize) {
+            Some(prev_receipt) => prev_receipt.cumulative_gas_used,
+            // if meta.index == 0, then there is no previous transaction
+            None => 0,
+        };
+
+        // gas_used = CumulativeGasUsed(tx_index) - CumulativeGasUsed(tx_index - 1)
+        res_receipt.gas_used = Some(
+            U256::from(receipt.cumulative_gas_used)
+                .checked_sub(U256::from(prev_tx_cumulative_gas_used))
```

Comment:
this can never overflow because cumulative_gas_used is always increasing, so we don't need to use checked math here.